address-space-controller/src/main/java/io/enmasse/controller/common/KubernetesHelper.java

```diff
@@ -122,12 +122,33 @@ public static boolean isStatefulSet(HasMetadata res) {
         return res.getKind().equals("StatefulSet");  // TODO: is there an existing constant for this somewhere?
     }
 
+    private void waitForZeroPods(String infraUuid, Map<String, String> labels, long timeoutInMillis) throws InterruptedException {
+        long end = System.currentTimeMillis() + timeoutInMillis;
+        int podsLeft = client.pods().withLabels(labels).list().getItems().size();
+        while (podsLeft > 0 && System.currentTimeMillis() < end) {
+            Thread.sleep(5000);
+            podsLeft = client.pods().withLabels(labels).list().getItems().size();
+        }
+        podsLeft = client.pods().withLabels(labels).list().getItems().size();
+        if (podsLeft > 0) {
+            String message = String.format("Failed to delete infra with uuid %s: %d pods still exists", infraUuid, podsLeft);
+            throw new RuntimeException(message);
+        }
+    }
+
     @Override
-    public void deleteResources(String infraUuid) {
+    public void deleteResources(String infraUuid) throws InterruptedException {
+        // Delete and await deployments to be deleted so that the per-address space controllers are not re-creating any resources
+        for (Deployment deployment : client.apps().deployments().withLabel(LabelKeys.INFRA_UUID, infraUuid).list().getItems()) {
+            if (!client.apps().deployments().delete(deployment)) {
```

Comment:
I think it might be safer to maintain `withPropagationPolicy("Background")` in the delete call.  Some parts of fabric8 switched to orphan dependants true (which would leave behind pods).    I find it difficult to remember which, so I think being explicit is the way to safe.  https://github.com/fabric8io/kubernetes-client/issues/1775

Reply:
@k-wall Fixed (though a bit clunky due to the api)